### ISSUE TITLE
Can't remove selected date in DatePicker

### ISSUE THREAD

**Issue by 4alexandr** [wix/wix-style-react#3043]
# 🐛 Bug Report

### 🏗 Relevant Components

DatePicker
...

### 😯 Current Behavior

After I choose any date I can't remove it. I can add "closeButton" to input, but click on this button will open DatePicker instead of removing current input value.
...

### 🤔 Expected Behavior

I should have the ability to remove selected date by "clear" button or by pressing "delete" button inside input.
...

### 👣 Steps to Reproduce
Just select a date and try to remove it.
https://codesandbox.io/s/2vm54qrx6j?module=%2Fsrc%2FApp%2Findex.js
  - **WSR version**: `6.6.1`

### 👀 Severity

Critical

**Comment by hepiyellow** [comment]
@wuwa @milkyfruit @LittleTal 
According to specs, there should not be a "clear (x) button",
but rather a cursor, with which (I guess) it is possible to clear the date.

Is there any special cursor behavior that is required?
Or it is simply a cursor that can delete any character of the Date? 
  Meaning that there will be a state where the date value is invalid, should there be any "invalid" visual indication?

**Comment by milkyfruit** [comment]
@4alexandr @hepiyellow 
The input itself can have "undefined" value, in this case it would display a placeholder's text, but the removing date shouldn't be part of component's UI.

The date removal should follow intention, like in Wix Events example date can be disabled with toggle "Date will be determined in a future (TBD)".

**Comment by hepiyellow** [comment]
@milkyfruit I Did not understand, can you rephrase...

**Comment by milkyfruit** [comment]
@hepiyellow Correct me if I'm wrong, this issue requests behaviour like this:
![image](https://user-images.githubusercontent.com/1684587/53478168-e263ea80-3a7e-11e9-853d-7e45ecbd8e1e.png)
I say, the behaviour should be like this:
![toggle](https://user-images.githubusercontent.com/1684587/53478323-3373de80-3a7f-11e9-9f8f-96cd10c22973.gif)
The clear field button shouldn't be part of DatePicker component, because the action must specify why the date is optional.

**Comment by 4alexandr** [comment]
@milkyfruit  DatePicket can be used in simple forms and even standalone, your example - complex form with own logic.
I try to use DatePicker in simple filter for a table, all I need - ability to set date and then - remove it without any additional UI.
Why standard html5 date input type allows me to remove selected date without any additional UI and wix-style-react no?

**Comment by hepiyellow** [comment]
@4alexander csn you share your ux spec?

**Comment by 4alexandr** [comment]
I don't have ux spec, I just want to have the ability to clear field value, as I can do with all other browser inputs, textarea...

**Comment by argshook** [comment]
@4alexandr hey, wix-style-react follows strict design guidelines and
according to them clear button is not part of DatePicker.

it may seem like obvious feature but it ain't so, believe it or not.

if you don't have ux spec, then im guessing you develop something that
is for internal use only.

if that's the case, then you can combine DatePicker with some button to
clear the value.

I'm sure you know how to do it, but just to illustrate, quick example:

```js
class MyComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state= {
      value: null
    }
  }

  onChange = value => this.setState({ value })


  clear() {
    this.setState({ value: null })
  }

  render() {
    return (
      <div>
        <DatePicker value={this.state.value} onChange={value => this.setState({ value })}/>
        <Button onClick={() => this.setState({ value: null })}>clear</Button>
      </div>
    );
  }
}
```

---

if you do, however, develop something user-facing, then you should have
a ux spec and then we should discuss this feature more broadly

**Comment by 4alexandr** [comment]
Yes, it's an internal feature and I can implement clear button(I already do it), but...
It's strange that to remove value from text input I should add additional button/UI.

DatePicker just a widget for text input. From my point of view - any widget should add/improve functionality/UX - not a reduce already existing.
Remove input value - base functionality for any input and if it missed in guidelines - it's bad guidelines.

You can add this functionality even without UI changes - just add keyboard support only for delete/backspace button.

**Comment by 4alexandr** [comment]
And one more - DatePicker allow me to use `inputProps`. It's a part of API.
Using `inputProps={{clearButton: true}}` I can show clear button inside input, but it doesn't work - click on this button just open DatePicker.
So it's a bug - need to fix clearButton or remove ability to add it.

**Comment by mykas** [comment]
Hi @4alexandr here is editor ux guidelines: https://v9yfnh.axshare.com/#g=1&p=date_picker . I think this case is a bit different then just a simple text input. Text Input can accept any entered value and no external modules depends on it. By clearing the value with the keyboard - calendar picker will need to handle wrong input values.

**Comment by 4alexandr** [comment]
Datepicker is a field. If field is not required - I should have ability to clear value.
`handle wrong input values` - or just fix `clear` button that appears with this DatePicker props: `inputProps={{clearButton: true}}`

**Comment by shlomitc** [comment]
closing as no decision was taken here and looks like a UX issue and not a technical one.
